websites/I/ITVX/presence.ts

```diff
@@ -0,0 +1,195 @@
+interface ChannelSlotData {
+	displayTitle: string;
+	start: string;
+	end: string;
+	seriesNumber?: number;
+	episodeNumber?: number;
+}
+
+interface ChannelMetadata {
+	id: string;
+	name: string;
+	slots: {
+		now: ChannelSlotData;
+		next: ChannelSlotData;
+	};
+	slug: string;
+}
+
+interface LiveChannelNextData {
+	props: {
+		pageProps: {
+			channelSlug: string;
+			channelsMetaData: {
+				channels: ChannelMetadata[];
+			};
+		};
+	};
+}
+
+interface CategoriesNextData {
+	props: {
+		pageProps: {
+			category: {
+				slug: string;
+				title: string;
+			};
+		};
+	};
+}
+
+interface ProgrammeNextData {
+	props: {
+		pageProps: {
+			title: {
+				CTAText?: string;
+				episodeNumber?: number;
+				programmeTitle: string;
+				seriesNumber?: number;
+				titleType: string;
+			};
+		};
+	};
+}
+
+enum Assets {
+	logo = "https://i.imgur.com/XisKvdg.png",
+	live = "https://i.imgur.com/WTcQZS7.png",
+	paused = "https://i.imgur.com/T6jkHHy.png",
+	playing = "https://i.imgur.com/1CbifM8.png",
+}
+
+const presence = new Presence({
+		clientId: "1043125926793318420",
+	}),
+	fetchLiveChannelNextData = (): LiveChannelNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	},
+	fetchCategoriesNextData = (): CategoriesNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	},
+	fetchProgrammeNextData = (): ProgrammeNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	};
```

Comment:
These should be 'proper' functions invoked with the `function` keyword instead of variable functions